packages/fx-core/src/plugins/solution/fx-solution/arm.ts

```diff
@@ -96,6 +105,156 @@ export async function generateArmTemplate(ctx: SolutionContext): Promise<Result<
   return ok(undefined); // Nothing to return when success
 }
 
+export async function deployArmTemplates(ctx: SolutionContext): Promise<Result<void, FxError>> {
+  const azureInfraDir = path.join(ctx.root, baseFolder);
+
+  // update parameters
+  const parameterTemplate = await fs.readFile(
+    path.join(azureInfraDir, parameterFolder, parameterTemplateFileName),
+    ConstantString.UTF8Encoding
+  );
+  const parameterJson = JSON.parse(expandParameterPlaceholders(ctx, parameterTemplate));
+  const parameterDefaultFilePath = path.join(
+    azureInfraDir,
+    parameterFolder,
+    parameterDefaultFileName
+  );
+  await fs.writeFile(parameterDefaultFilePath, parameterJson);
+  const resourceGroupName = parameterJson.parameters.resourceBaseName;
+  if (!resourceGroupName) {
+    throw returnSystemError(
+      new Error("Failed to get resource group from parameters."),
+      PluginNames.SOLUTION,
+      SolutionError.NoResourceGroupFound
+    );
+  }
+
+  // Compile bicep file to json
+  const orchestrationFilePath = path.join(
+    azureInfraDir,
+    templateFolder,
+    bicepOrchestrationFileName
+  );
+  const armTemplateJsonFilePath = path.join(azureInfraDir, templateFolder, armTemplateJsonFileName);
+  await compileBicepToJson(orchestrationFilePath, armTemplateJsonFilePath);
+  ctx.logProvider?.info("Successfully compile bicep files to JSON.");
+
+  // deploy arm templates to azure
+  const client = await getResourceManagementClientForArmDeployment(ctx);
+  const deploymentName = `${PluginNames.SOLUTION}-deployment`;
+  const deploymentParameters: ResourceManagementModels.Deployment = {
+    properties: {
+      parameters: parameterJson,
+      template: await fs.readFile(armTemplateJsonFilePath, ConstantString.UTF8Encoding),
+      mode: "Incremental" as ResourceManagementModels.DeploymentMode,
+    },
+  };
+  let deploymentFinished = false;
+  try {
+    const result = await client.deployments
+      .createOrUpdate(resourceGroupName, deploymentName, deploymentParameters)
+      .then((result) => {
+        ctx.logProvider?.info(
+          `Successfully deploy arm templates to Azure. Resource group name: ${resourceGroupName}. Deployment name: ${deploymentName}`
+        );
+        return result;
+      })
+      .finally(() => {
+        deploymentFinished = true;
+      });
+    pollDeploymentStatus(client, resourceGroupName, Date.now());
+    if (!ctx.projectSettings?.solutionSettings) {
+      return err(
+        returnSystemError(
+          new Error("solutionSettings is undefined"),
+          PluginNames.SOLUTION,
+          SolutionError.InternelError
+        )
+      );
+    }
+    ctx.projectSettings.solutionSettings["armTemplateOutput"] = result.properties?.outputs;
```

Comment:
should use `ctx.configs.get('solution').set('armTemplateOutput',...)`. Existing code is operation the settings.json

Reply:
Fixed.